src/interf.jl

```diff
@@ -50,12 +66,16 @@ function nnmf(X::AbstractMatrix{T}, k::Integer;
 
     # run optimization
     ret = solve!(alginst, X, W, H)
-    
+
+    # replicates
+    minobjv = ret.objvalue
     for _ in 2:replicates
-        if ret.converged
-            break
+        W, H = randinit(X, k; zeroh=!initH, normalize=true)
```

Comment:
Doesn't this overwrite the provided custom W and H?

Reply:
Yes, I'll fix the commit.

Reply:
What you had before, setting `W` and `H` to `W0` and `H0`, respectively, without copying was okay. What I mean is that in this loop you're initializing new `W` and `H` matrices rather than using the ones that have already been defined. Is that what you intended?

Reply:
Yes, `replicates` in MATLAB uses "new" random initial values for W and H at each replication, except at the first replication specified by `init`. So, in this loop I'm initializing new W and H matrices. The latest commit intends to prevent `nnmf` from overwriting the arguments `W0` and `H0`.

Reply:
I see, okay. I think the behavior that avoids copying should be preferred though, with a note in the documentation that `W0` and `H0` may be overwritten. The user can pass in copies themselves if they need to.

Reply:
Sounds good. I'll add the following note to the first commit. 

**Note:** ``W0`` and ``H0`` may be overwritten. If one needs to avoid it, please pass in copies themselves.